cmd/main.go

```diff
@@ -122,6 +123,7 @@ func cmd() *cobra.Command {
 	c.Flags().StringVar(&opts.watchNamespace, "watch-namespace", "", "Namespace to watch for Kubernetes resources. Defaults to all namespaces.")
 	c.Flags().StringVar(&opts.managerName, "manager-name", "ngrok-ingress-controller-manager", "Manager name to identify unique ngrok ingress controller instances")
 	c.Flags().BoolVar(&opts.useExperimentalGatewayAPI, "use-experimental-gateway-api", false, "sets up experemental gatewayAPI")
+	c.Flags().StringVar(&opts.clusterDomain, "cluster-domain", "svc.cluster.local", "Cluster domain used in the cluster")
```

Comment:
My only nit for this is this default seems to be declared in a few places and some of the defaults might be overkill. 
The helm values has a default of `clusterDomain: svc.cluster.local`

The template checks if it has a value before passing it, but would only be false if it was overridden to "" via
```
        {{- if .Values.clusterDomain }}
        - --cluster-domain={{ .Values.clusterDomain }}
        {{- end }}
```

But if this isn't passed it then gets the default from this flag. 

Lastly the driver then also has the `const defaultClusterDomain = "svc.cluster.local"` which is then overridden in this main.go via `WithClusterDomain`

It's not the end of the world, and I don't think the system would work if it was set to empty string or something, but it just seemed like a lot of layers to grok.

Reply:
Part of this is my fault when merging in main and noticing we have quite a few named arguments to `NewDriver`. I did a small functional options pattern refactor but still feel like having it default in the driver is reasonable. Since we also have it as a command line flag for now, I'm also okay with having it there so you can easily see the default when running `go run ./cmd/main.go --help`